### PR TITLE
$response object passed as service argument instead of resolveDispositionHeaderFilename() argument

### PR DESCRIPTION
**Pull request by inmarelibero**
$request object is now passed through services.xml instead of passing it optionally in the controller as option.
What do you think?

### CODE REVIEW

**Comment by igorw** [comment]
I wanted to avoid scoping everything to request, because it's messy. But it's probably a good idea for convenience.

File: Response/AbstractResponseFactory.php [hunk 1]
```diff
@@ -7,13 +7,15 @@
 abstract class AbstractResponseFactory
 {
     protected $baseDir;
+    protected $request;
     protected $fullFilename;
     protected $contentType;
     protected $options;
 
-    public function __construct($baseDir)
+    public function __construct($baseDir, $request)
```
Comment (igorw) [inline Response/AbstractResponseFactory.php]: Can you add a type hint for `Request`?

File: Response/AbstractResponseFactory.php [hunk 2]
```diff
@@ -59,17 +60,16 @@ protected function resolveDispositionHeader(array $options)
...
     }
 
     /**
      * Algorithm inspired by phpBB3
      */
-    protected function resolveDispositionHeaderFilename($filename, $request)
+    protected function resolveDispositionHeaderFilename($filename)
     {
-        $userAgent = !is_null($request) ? $request->headers->get('User-Agent') : null;
+        $userAgent = !is_null($this->request) ? $this->request->headers->get('User-Agent') : null;
```
Comment (igorw) [inline Response/AbstractResponseFactory.php:74]: The `is_null` is no longer needed. And neither is the `if (!$userAgent)` below.

Comment (inmarelibero) [inline Response/AbstractResponseFactory.php:74]: Ok, I was not sure if in some circumstances `$this->request` could be null. I remove it

File: Response/AbstractResponseFactory.php (hunk 1 above)
Comment (inmarelibero) [inline Response/AbstractResponseFactory.php]: yes, i put a `use Symfony\Component\HttpFoundation\Request;` in the top, like `use Symfony\Component\HttpFoundation\Response;`

**Comment by inmarelibero** [comment]
yes, I found a bit complicate for users to be forced to pass the `$request` object in the `create()`function just to avoid the problem in IE, Safari and Konqueror. And that it should be transparent for the user.

**Comment by igorw** [comment]
No, the default behaviour is to send the invalid header which works in all browsers. You need to pass the request to get the valid one.

**Comment by igorw** [comment]
Can you adjust the README?

**Comment by inmarelibero** [comment]
just to make it clear:
- now users are not required to pass the $request object in the controller as option to the `create()` function
- I update the readme writing that support for browsers that are not RFC compliant (IE, safari and konqueror) has been provided

right?

**Comment by igorw** [comment]
I've updated it myself.

Ist gemerged ;-)

**Comment by inmarelibero** [comment]
ok cool. many thanks for your work!

**Pull request merged**